# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/quota/ActorsParent.cpp
# Commit: c8a45415dc5b
# Full Hash: c8a45415dc5b0b33b7e9d4b9cd8114ac09a05218
# Author: Jens Stutte <jstutte@mozilla.com>
# Date: 2021-05-28 21:48:03
# Description:
#   Bug 1706998: Replace QuotaManager::GetRef().MaybeRecordShutdownStep by static QuotaManager::(Safe)MaybeRecordQuotaClientShutdownStep; r=dom-storage-reviewers,janv
#   
#   There are some few unsafe uses of MaybeRecordShutdownStep where the QuotaManager singleton may be  not (yet or anymore) alive.
#   In order to not add many unnecessary null checks, we drop GetRef and MaybeRecordShutdownStep in favor of
#   QuotaManager::MaybeRecordQuotaClientShutdownStep
# ==============================================================================

diff -r cc21fb9be25a -r c8a45415dc5b dom/quota/ActorsParent.cpp
--- a/dom/quota/ActorsParent.cpp	Fri May 28 05:49:26 2021 +0000
+++ b/dom/quota/ActorsParent.cpp	Fri May 28 06:46:04 2021 +0000
@@ -3285,13 +3285,6 @@
 }
 
 // static
-QuotaManager& QuotaManager::GetRef() {
-  MOZ_ASSERT(gInstance);
-
-  return *gInstance;
-}
-
-// static
 bool QuotaManager::IsShuttingDown() { return gShutdown; }
 
 // static
@@ -3640,11 +3633,14 @@
   return NS_OK;
 }
 
-void QuotaManager::MaybeRecordShutdownStep(const Client::Type aClientType,
-                                           const nsACString& aStepDescription) {
+// static
+void QuotaManager::SafeMaybeRecordQuotaClientShutdownStep(
+    const Client::Type aClientType, const nsACString& aStepDescription) {
   // Callable on any thread.
 
-  MaybeRecordShutdownStep(Some(aClientType), aStepDescription);
+  if (auto* const quotaManager = QuotaManager::Get()) {
+    quotaManager->MaybeRecordShutdownStep(Some(aClientType), aStepDescription);
+  }
 }
 
 void QuotaManager::MaybeRecordQuotaManagerShutdownStep(